org/mozilla/jss/tests/BenchmarkSSLSocket.java

```diff
@@ -0,0 +1,245 @@
+ *     implementations will utilize a nickname instead.
+ *  3. The port to listen on.
+ *  4. The size of the HTTP message to fake.
+ *
+ * Note that, when utilizing a JSS provider, JSS must be loaded via a
+ * java.security. When utilizing SunJSSE, for best results, do not load
+ * JSS via java.security.
+ *
+ * It is suggested to use run_test.sh from the build/ directory for this.
+ */
+public class BenchmarkSSLSocket {
+    public String type;
+    public int port;
+    public int size;
+    public String nickname;
+
+    public String headers = "HTTP/1.1 200 OK\r\nConnection: Closed\r\n";
+    public String message;
+
+    public BenchmarkSSLSocket(String[] args) throws Exception {
+        type = args[0];
+        nickname = args[1];
+        port = Integer.parseInt(args[2]);
+        size = Integer.parseInt(args[3]);
+
+        headers = headers + "Content-Length: " + size + "\r\n";
+
+        StringBuilder sb = new StringBuilder(size);
+        for (int i = 0; i < size; i++) {
+            sb.append("a");
+        }
+
+        message = headers + "\r\n" + sb.toString();
+    }
+
+    public ServerSocket getServerSocket() throws Exception {
+        switch (type) {
+            case "JSS.legacy": {
+                org.mozilla.jss.ssl.SSLServerSocket sock = new org.mozilla.jss.ssl.SSLServerSocket(port);
+                sock.setSoTimeout(0);
+                org.mozilla.jss.ssl.SSLServerSocket.configServerSessionIDCache(0, 43200, 43200, null);
+
+                sock.setReuseAddress(true);
+                sock.requestClientAuth(false);
+                sock.requireClientAuth(org.mozilla.jss.ssl.SSLSocket.SSL_REQUIRE_NEVER);
+                sock.setUseClientMode(false);
+                sock.setServerCertNickname(nickname);
+
+                return sock;
+            }
+            case "JSS.SSLSocket": {
+                KeyManagerFactory kmf = KeyManagerFactory.getInstance("NssX509");
+
+                SSLContext ctx = SSLContext.getInstance("TLS", "Mozilla-JSS");
+                ctx.init(
+                    kmf.getKeyManagers(),
+                    new TrustManager[] { new JSSNativeTrustManager() },
+                    null
+                );
+
+                SSLServerSocketFactory factory = ctx.getServerSocketFactory();
+                org.mozilla.jss.ssl.javax.JSSServerSocket sock = (org.mozilla.jss.ssl.javax.JSSServerSocket) factory.createServerSocket(port);
+
+                sock.setReuseAddress(true);
+                sock.setWantClientAuth(false);
+                sock.setNeedClientAuth(false);
+                sock.setUseClientMode(false);
+                sock.setCertFromAlias(nickname);
+
+                return sock;
+            }
+            case "SunJSSE.SSLSocket": {
+                FileInputStream fis = new FileInputStream(nickname);
+                KeyStore store = KeyStore.getInstance("PKCS12");
+                store.load(fis, "m1oZilla".toCharArray());
```

Comment:
The PKCS #12 password is hardcoded, so we should mention that in the doc as well, or provide a param to specify the password.

Reply:
Ah, I've added an optional form for the password, thanks!